### PR TITLE
Test/ Add constructor and initialize assessment tests

### LINKED ISSUE #432: Assessment unit tests: constructor & initialize
**constructor**

- [ ] Should set nxm address correctly

**initialize**

- [x] Should revert if already initialized
- [ ] Should set the config parameters
- [ ] Should call token controller to add the address to whitelist

### PR DESCRIPTION
**Pull request by paterson1**
## Context

Closes #432 

## Changes proposed in this pull request

Adds new unit tests for the constructor and initialize. Implemented the list of test cases from the issue. `nxm` was changed to `public` to be able to fetch it for testing, and logic to test whitelisting was added to `ASMockTokenController`. 

## Test plan

New tests were added in `test/unit/Assessment/initialize.js` and `test/unit/Assessment/constructor.js`

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

**Comment by danoctavian** [comment]
@paterson1  conflicts with base branch - needs a rebase

File: test/unit/Assessment/initialize.js [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
+
+    expect(preInitializeConfig.minVotingPeriodInDays).to.be.equals(0);
+    expect(preInitializeConfig.stakeLockupPeriodInDays).to.be.equals(0);
+    expect(preInitializeConfig.payoutCooldownInDays).to.be.equals(0);
+    expect(preInitializeConfig.silentEndingPeriodInDays).to.be.equals(0);
+
+    await assessment.initialize();
+    const afterInitializeConfig = await assessment.config();
+
+    expect(afterInitializeConfig.minVotingPeriodInDays).to.be.to.be.equals(3);
```
Comment (danoctavian) [inline test/unit/Assessment/initialize.js:31]: to.be.to.be repeated

interestingly it works.

File: test/unit/Assessment/initialize.js [hunk 2]
```diff
@@ -1,8 +1,51 @@
...
+    const Assessment = await ethers.getContractFactory('Assessment');
+    const assessment = await Assessment.deploy(nxm.address);
+
+    await master.setLatestAddress(hex('AS'), assessment.address);
+    await assessment.changeMasterAddress(master.address);
+    await assessment.changeDependentContractAddress();
+
+    const preInitializeConfig = await assessment.config();
+
+    expect(preInitializeConfig.minVotingPeriodInDays).to.be.equals(0);
```
Comment (danoctavian) [inline test/unit/Assessment/initialize.js:23]: write `to.be.equal` (singular) not `equals`

File: test/unit/Assessment/castVotes.js [hunk 3]
```diff
@@ -314,4 +314,154 @@ describe('castVotes', function () {
       await expect(tx).to.emit(assessment, 'VoteCast').withArgs(user2.address, 1, parseEther('1000'), false);
```
Comment (danoctavian) [inline test/unit/Assessment/castVotes.js:314]: these tests belong to another PR right?

File: contracts/mocks/Assessment/ASMockTokenController.sol [hunk 4]
```diff
@@ -21,8 +24,12 @@ contract ASMockTokenController {
     token.mint(_to, _value);
   }
 
-  function addToWhitelist(address) external returns (bool) {
-    // no-op
+  function addToWhitelist(address _member) public {
+    addToWhitelistLastCalledWith = _member;
+  }
+
+  function removeFromWhitelist(address _member) public {
```
Comment (danoctavian) [inline contracts/mocks/Assessment/ASMockTokenController.sol:31]: this doesn't seem to be used.

No test for removal.

removeFromWhitelist

**Review by danoctavian — CHANGES_REQUESTED** [review]
Some changes necessary

**Review by danoctavian — APPROVED** [review]
LGTM

**Review by roxdanila — APPROVED** [review]

**Pull request merged**